app/src/androidTest/java/com/github/se/polyfit/meal/MealViewModelTest.kt

```diff
@@ -0,0 +1,148 @@
+package com.github.se.polyfit.meal
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.github.se.polyfit.data.repository.MealRepository
+import com.github.se.polyfit.model.ingredient.Ingredient
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.model.nutritionalInformation.MeasurementUnit
+import com.github.se.polyfit.model.nutritionalInformation.Nutrient
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+import io.mockk.mockk
+import java.time.LocalDate
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class MealViewModelTest {
+
+  @get:Rule val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+  private lateinit var viewModel: MealViewModel
+  private val mealRepo = mockk<MealRepository>(relaxed = true)
+  private val context = mockk<android.content.Context>(relaxed = true)
+
+  @Before
+  fun setup() {
+    viewModel = MealViewModel(mealRepo, Meal.default(), "firebaseID")
+  }
+
+  @Test
+  fun setMealName_updates_meal_name_and_completeness() = runBlocking {
+    val mealName = "Test Meal"
+    viewModel.setMealData(Meal.default())
+    viewModel.setMealName(mealName)
+
+    val updatedMeal = viewModel.meal.getOrAwait()
```

Comment:
Maybe rename getOrAwait to waitData

Reply:
The naming comes from the fact that if the data is already available, there is no need to wait; it directly returns. If the live data has not been updated, then it waits for a certain amount of time (3 seconds in this case). If the data has not changed while waiting, then throw an error.